File: src/app/dto/UserDTO.ts
import {EmployeeDTO} from './EmployeeDTO';
import {RoleDTO} from './RoleDTO';

export class UserDTO {
  id!: number | null;
  emailAddress!: string;
  roles!: RoleDTO[];
  employee!: EmployeeDTO;

  constructor() {
    this.roles = [new RoleDTO()];
    this.employee = new EmployeeDTO;
  }
}

File: src/main/java/teampearl/cvgenerator/back_end/presentation_layer/dto/UserDTO.java
package teampearl.cvgenerator.back_end.presentation_layer.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class UserDTO {
    @JsonProperty("id")
    private Integer id;

    @JsonProperty("emailAddress")
    private String emailAddress;

    @JsonProperty("roles")
    private List<RoleDTO> roles;

    @JsonProperty("employee")
    private EmployeeDTO employee;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getEmailAddress() {
        return emailAddress;
    }

    public void setEmailAddress(String emailAddress) {
        this.emailAddress = emailAddress;
    }

    public List<RoleDTO> getRoles() {
        return roles;
    }

    public void setRoles(List<RoleDTO> roles) {
        this.roles = roles;
    }

    public EmployeeDTO getEmployee() {
        return employee;
    }

    public void setEmployee(EmployeeDTO employee) {
        this.employee = employee;
    }
}
